internal/engine/ingester/deps/deps.go

```diff
@@ -0,0 +1,225 @@
+// SPDX-FileCopyrightText: Copyright 2024 The Minder Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package deps provides the deps rule data ingest engine
+package deps
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/go-git/go-billy/v5"
+	"github.com/go-git/go-billy/v5/helper/iofs"
+	"github.com/go-viper/mapstructure/v2"
+	scalibr "github.com/google/osv-scalibr"
+	"github.com/google/osv-scalibr/extractor/filesystem/list"
+	scalibr_fs "github.com/google/osv-scalibr/fs"
+	scalibr_plugin "github.com/google/osv-scalibr/plugin"
+	"github.com/google/uuid"
+	"github.com/protobom/protobom/pkg/sbom"
+	"github.com/rs/zerolog"
+	"google.golang.org/protobuf/reflect/protoreflect"
+
+	engerrors "github.com/mindersec/minder/internal/engine/errors"
+	pb "github.com/mindersec/minder/pkg/api/protobuf/go/minder/v1"
+	"github.com/mindersec/minder/pkg/engine/v1/interfaces"
+	"github.com/mindersec/minder/pkg/entities/v1/checkpoints"
+	provifv1 "github.com/mindersec/minder/pkg/providers/v1"
+)
+
+const (
+	// DepsRuleDataIngestType is the type of the deps rule data ingest engine
+	DepsRuleDataIngestType = "deps"
+	defaultBranch          = "main"
+)
+
+// Deps is the engine for a rule type that uses deps data ingest
+type Deps struct {
+	cfg     *pb.DepsType
+	gitprov provifv1.Git
+}
+
+// Config is the set of parameters to the deps rule data ingest engine
+type Config struct {
+	Branch string `json:"branch" yaml:"branch" mapstructure:"branch"`
+}
+
+// NewDepsIngester creates a new deps rule data ingest engine
+func NewDepsIngester(cfg *pb.DepsType, gitprov provifv1.Git) (*Deps, error) {
+	if gitprov == nil {
+		return nil, fmt.Errorf("provider is nil")
+	}
+
+	if cfg == nil {
+		cfg = &pb.DepsType{}
+	}
+
+	return &Deps{
+		cfg:     cfg,
+		gitprov: gitprov,
+	}, nil
+}
+
+// GetType returns the type of the git rule data ingest engine
+func (*Deps) GetType() string {
+	return DepsRuleDataIngestType
+}
+
+// GetConfig returns the config for the git rule data ingest engine
+func (gi *Deps) GetConfig() protoreflect.ProtoMessage {
+	return gi.cfg
+}
+
+// Ingest does the actual data ingestion for a rule type by cloning a git repo,
+// and scanning it for dependencies with scalibr.
+func (gi *Deps) Ingest(ctx context.Context, ent protoreflect.ProtoMessage, params map[string]any) (*interfaces.Result, error) {
+	switch entity := ent.(type) {
+	case *pb.Repository:
+		return gi.ingestRepository(ctx, entity, params)
+	default:
+		return nil, fmt.Errorf("deps is only supported for repositories")
+	}
+}
+func (gi *Deps) ingestRepository(ctx context.Context, repo *pb.Repository, params map[string]any) (*interfaces.Result, error) {
```

Comment:
`nit`: I wonder if we could call the `git` ingester directly from here and THEN process the result of that.

Reply:
Sgmt, I'll refactor it in follow up.

Reply:
yeah, doesn't need to be in this PR